apis/network/v1beta1/openstacknet_types.go

```diff
@@ -0,0 +1,153 @@
+/*
+Copyright 2023.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	"github.com/openstack-k8s-operators/lib-common/modules/common/condition"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// OpenStackNetSpec defines the desired state of OpenStackNet
+type OpenStackNetSpec struct {
+	// +kubebuilder:validation:Optional
+	//  List of subnets
+	Subnets []OpenStackSubnet `json:"subnets,omitempty"`
```

Comment:
wondering if we need this distinction openstacknet -> opernstacksubnet list where a net has multiple subnets? or wouldn't it be enough to just have openstacknet?

Reply:
You mean assume one net one subnet?  What about https://github.com/openstack/tripleo-heat-templates/blob/master/network-data-samples/routed-networks.yaml and dual stack?